static/app/utils/replays/hooks/useReplayReader.tsx

```diff
@@ -0,0 +1,48 @@
+import {useMemo} from 'react';
+
+import useReplayData from 'sentry/utils/replays/hooks/useReplayData';
+import ReplayReader from 'sentry/utils/replays/replayReader';
+
+type Props = {
+  orgSlug: string;
+  replaySlug: string;
+};
+
+export default function useReplayReader({orgSlug, replaySlug}: Props) {
+  const replayId = parseReplayId(replaySlug);
+
+  const {attachments, errors, replayRecord, ...replayData} = useReplayData({
+    orgSlug,
+    replayId,
+  });
+
+  const replay = useMemo(
+    () => ReplayReader.factory({attachments, errors, replayRecord}),
+    [attachments, errors, replayRecord]
+  );
+
+  return {
+    ...replayData,
+    attachments,
+    errors,
+    replay,
+    replayId,
+    replayRecord,
+  };
+}
+
+// see https://github.com/getsentry/sentry/pull/47859
+// replays can apply to many projects when incorporating backend errors
+// this makes having project in the `replaySlug` obsolete
+// we must keep this url schema for now for backward compat but we should remove it at some point
+// TODO: remove support for projectSlug in replay url?
```

Comment:
Do we know when we can remove it?

Reply:
I don't think any urls have been updated to remove the projectSlug yet. So no date set :(

Reply:
ticketed: https://github.com/getsentry/team-replay/issues/80

I'll put this on a list of possible onboarding tasks